src/kernels/execution/helpers.ts

```diff
@@ -784,10 +791,14 @@ export async function updateNotebookMetadata(
                         hash: await getInterpreterHash(kernelConnection.interpreter)
                     }
                 };
-                if ('interpreter' in metadata) {
-                    delete metadata['interpreter'];
+            } else {
+                if ('vscode' in metadata) {
+                    delete metadata['vscode'];
                 }
             }
+            if ('interpreter' in metadata) {
+                delete metadata['interpreter'];
+            }
```

Comment:
The if is redundant:

```suggestion
            } else {
                delete metadata['vscode'];
            }
            delete metadata['interpreter'];
```